public/drawImageWorker.js

```diff
@@ -71,12 +68,23 @@ function processScalePoints(scalePoints) {
 
     if(sharedArray) sharedArray[i] = gammaCorrected
 
+    const outputImageData = outputImage.data
     const j = i * 4
+
     outputImageData[j] = gammaCorrected
     outputImageData[j + 1] = gammaCorrected
     outputImageData[j + 2] = gammaCorrected
     outputImageData[j + 3] = 255
   }
-  context.putImageData(outputImage, 0, 0)
-  postMessage({'updateSharedArray': true})
+
+  await context.putImageData(outputImage, 0, 0)
```

Comment:
This should also be called only for the non RGB case right?

Reply:
wait nvm i guess we are using the canvas there too. But this doesn't return a promise right? So await doesn't mean anything here?